Define THPStorage struct only once (rather than N times)

lhuang04 opened the pull request:
Summary:
The definetion of THPStorage does not depend on any Real, its macro
defintion is unnecessary, refactor the code so that THPStorage is not macro
defined.

Differential Revision: D13340445

ezyang commented on torch/csrc/generic/Storage.h:
```diff
@@ -2,10 +2,13 @@
 #else
 
+#ifndef TH_GENERIC_STORAGE_DEF
+#define TH_GENERIC_STORAGE_DEF
```
Strongly encourage you to just move this out of this header entirely. Then, you can have `torch/csrc/Storage.h` include whatever header you decide to put it in, so you don't have to fix include sites. (And it will only be included once, via usual `pragma once` or include guard shenanigans)

lhuang04 committed:
Define THPStorage struct only once (rather than N times) (#14802)

Summary:
Pull Request resolved: https://github.com/pytorch/pytorch/pull/14802

The definetion of THPStorage does not depend on any Real, its macro
defintion is unnecessary, refactor the code so that THPStorage is not macro
defined.

Reviewed By: ezyang

Differential Revision: D13340445

fbshipit-source-id: 226e4e2670b17421d2b2fc7cbb85b2a772812c10
